Monika After Story/game/zz_hangman.rpy

```diff
@@ -724,10 +675,110 @@ label mas_hangman_game_loop:
 
         "No.":
             pass
+    jump mas_hangman_game_end
 
-            #FALL THROUGH
+label mas_hangman_game_win(guesses=0, chances=0, last_chance_guesses=0):
+    if is_window_sayori_visible:
+        show hm_s_win_6 as window_sayori at hangman_sayori_h
+
+    if player_word:
+        $ the_word = "your name"
+    else:
+        $ the_word = "the word"
+        
+    if chances == 6:
+        m 1sud "Wow, [player]..."
+        if not player_word:
+            m 1hua "You didn't miss a single letter!"
+            m 3eub "You must have a really good vocabulary, ahaha."
+            m 1tsa "...Or maybe you just lucked out, hmm?"
+        else:
+            m 1hkb "You just guessed your own name flawlessly, ahaha."
+    elif chances == 1 and last_chance_guesses / (guesses - (6 - chances)) > 0.5:
```

Comment:
since `last_change_guesses / (guesses - (6 - chances)) > 0.5` is used elsewhere and is not obvious what this means without the comment, can you make this a function with an understandable name? maybe `lastChanceGuessPct`

Reply:
Sure, no problem, thanks for telling me.